concent_api/core/admin.py

```diff
@@ -18,22 +18,25 @@ class ActivePassiveDownloadsStateFilter(admin.SimpleListFilter):
 
     def lookups(self, request, model_admin):
         return (
-            ('active', 'Active'),
-            ('passive', 'Passive'),
-            ('active_or_downloads', 'Active or with active downloads')
+            ('active', 'Active and not timed out '),
+            ('passive', 'Passive and timed out'),
+            ('active_or_downloads', 'Active and not timed out or with active downloads')
         )
 
     def queryset(self, request, queryset: QuerySet) -> QuerySet:
         assert self.value() in {'active', 'passive', 'active_or_downloads', None}
+        current_timestamp = get_current_utc_timestamp()
         if self.value() == 'active':
-            return queryset.filter(state__in=ACTIVE_STATE_NAMES)
+            return queryset.filter(Q(state__in=ACTIVE_STATE_NAMES, next_deadline__gte=parse_timestamp_to_utc_datetime(current_timestamp)))
```

Comment:
`Q` is not needed here, you can use several conditions in `.filter` unless you want to make them sql `OR` (`|`), and they will work as sql `AND` (this is Django default).
Please note that in `.exclude` it works differently.

Reply:
Done.